Generate projects for Java 11 by default

gsmet opened the pull request:
@gastaldi I created the PR early so that you can have a look too.

I will try generating projects now :).

gsmet committed:
Generate projects for Java 11 by default

gastaldi reviewed (approved):
LGTM

gsmet commented:
OK, I tested all the project creation combinations, it was fun \o/.

gsmet commented:
@patriot1burke should we also make this change for your Amazon Lambda archetypes? Or do you want to stick to 8 for now?

gastaldi commented:
You need to remove the `Quarkus CI / JDK 8 JVM Tests` from the GH Actions descriptor too

gsmet committed:
Parameterize the Java target

By default, it is based on the Java version used to create the project.
It can be overridden in CreateProject though if you really want to
enforce it in your tools.

I haven't exposed it for now as we will probably remove it or tweak it
later.

Note that even if you enforce it, it will still use either 8 or 11 for
now. We might want to further enhance that later.

gsmet commented:
@gastaldi I don't want to do that. We want to support 8 still. But I think I pushed an appropriate work around.

geoand commented on independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/CreateProject.java:
```diff
@@ -134,6 +147,15 @@ public boolean doCreateProject(final Map<String, Object> context) throws IOExcep
     }
 
     public QuarkusCommandOutcome execute() throws QuarkusCommandException {
+        // Define the Java version to use determined from the one specified or the one creating the project
```
The comment isn't really helpful, just saying :)

patriot1burke commented:
Yes I'll do that now with my other work.

On Wed, Mar 25, 2020 at 12:47 PM Georgios Andrianakis <
notifications@github.com> wrote:

> *@geoand* commented on this pull request.
> ------------------------------
>
> In
> independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/CreateProject.java
> <https://github.com/quarkusio/quarkus/pull/8151#discussion_r398007773>:
>
> > @@ -134,6 +147,15 @@ public boolean doCreateProject(final Map<String, Object> context) throws IOExcep
>      }
>
>      public QuarkusCommandOutcome execute() throws QuarkusCommandException {
> +        // Define the Java version to use determined from the one specified or the one creating the project
>
> The comment isn't really helpful, just saying :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/quarkusio/quarkus/pull/8151#pullrequestreview-381338836>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFL53YM5W5ZHN546V3OKULRJIYR3ANCNFSM4LTPTECA>
> .
>


-- 
Bill Burke
Red Hat

maxandersen commented on independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/CreateProject.java:
```diff
@@ -134,6 +147,15 @@ public boolean doCreateProject(final Map<String, Object> context) throws IOExcep
+        Matcher matcher = JAVA_VERSION_PATTERN
+                .matcher(this.javaTarget != null ? this.javaTarget : System.getProperty("java.version", ""));
+        if (matcher.matches() && Integer.parseInt(matcher.group(1)) < 11) {
+            invocation.setProperty(JAVA_TARGET, invocation.getBuildTool() == BuildTool.MAVEN ? "1.8" : "1_8");
```
why not have one "java_target" and another "java_underscore_target" so the same property isn't to be treated differently ?

gsmet replied:
That was the other option I pondered. I decided to go with this one.

maxandersen commented on devtools/platform-descriptor-json/src/main/resources/templates/basic-rest/kotlin/pom.xml-template.ftl:
```diff
@@ -107,7 +107,7 @@
                 <configuration>
                     <javaParameters>true</javaParameters>
-                    <jvmTarget>1.8</jvmTarget>
+                    <jvmTarget>${java_target}</jvmTarget>
```
so if grok this right this change is a change in the "api" of the code generation that means the templates now *require* java_target to be set which will not be the case if you are using 1.3.0 plugin (@aloubyansky correct me if i'm wrong on this here).

So 1.3.0 plugin will pick up latest 1.3.x platform and when generation happens it will fail.

To remedy this the various .ftl files should check if java_target is set.
if my freemarker syntax memory is right its something like `${java_target!"11"}`

aloubyansky replied:
> so if grok this right this change is a change in the "api" of the code generation that means the templates now _require_ java_target to be set which will not be the case if you are using 1.3.0 plugin (@aloubyansky correct me if i'm wrong on this here).

That is correct.

gsmet replied:
I suppose we should choose 1.8 as the default to have a consistent behavior. I'll go do that and test.

gsmet committed:
Keep template files compatible with Maven plugin < 1.3.1

maxandersen commented:
I approve despite I feel dirty doing so :)

gsmet commented:
@maxandersen yeah... and you're not the one who had to write this :).
